pandas/tests/resample/test_datetime_index.py

```diff
@@ -25,1484 +23,1491 @@
+    assert result.iloc[0] == s['1/4/2005']
+    assert result.iloc[1] == s['1/10/2005']
+
+    result = s.resample('W-WED').last()
+    assert len(result) == 2
+    assert (result.index.dayofweek == [2, 2]).all()
+    assert result.iloc[0] == s['1/5/2005']
+    assert result.iloc[1] == s['1/10/2005']
+
+    result = s.resample('W-THU').last()
+    assert len(result) == 2
+    assert (result.index.dayofweek == [3, 3]).all()
+    assert result.iloc[0] == s['1/6/2005']
+    assert result.iloc[1] == s['1/10/2005']
+
+    result = s.resample('W-FRI').last()
+    assert len(result) == 2
+    assert (result.index.dayofweek == [4, 4]).all()
+    assert result.iloc[0] == s['1/7/2005']
+    assert result.iloc[1] == s['1/10/2005']
+
+    # to biz day
+    result = s.resample('B').last()
+    assert len(result) == 7
+    assert (result.index.dayofweek == [4, 0, 1, 2, 3, 4, 0]).all()
+
+    assert result.iloc[0] == s['1/2/2005']
+    assert result.iloc[1] == s['1/3/2005']
+    assert result.iloc[5] == s['1/9/2005']
+    assert result.index.name == 'index'
+
+
+def test_resample_upsampling_picked_but_not_correct():
+
+    # Test for issue #3020
+    dates = date_range('01-Jan-2014', '05-Jan-2014', freq='D')
+    series = Series(1, index=dates)
+
+    result = series.resample('D').mean()
+    assert result.index[0] == dates[0]
+
+    # GH 5955
+    # incorrect deciding to upsample when the axis frequency matches the
+    # resample frequency
+
+    import datetime
+    s = Series(np.arange(1., 6), index=[datetime.datetime(
+        1975, 1, i, 12, 0) for i in range(1, 6)])
+    expected = Series(np.arange(1., 6), index=date_range(
+        '19750101', periods=5, freq='D'))
+
+    result = s.resample('D').count()
+    assert_series_equal(result, Series(1, index=expected.index))
 
-        b = TimeGrouper(Minute(5), closed='right', label='right')
-        g = s.groupby(b)
-        # check all cython functions work
-        funcs = ['add', 'mean', 'prod', 'ohlc', 'min', 'max', 'var']
-        for f in funcs:
-            g._cython_agg_general(f)
+    result1 = s.resample('D').sum()
+    result2 = s.resample('D').mean()
+    assert_series_equal(result1, expected)
+    assert_series_equal(result2, expected)
 
-        assert g.ngroups == 2593
-        assert notna(g.mean()).all()
 
-        # construct expected val
-        arr = [1] + [5] * 2592
-        idx = dti[0:-1:5]
-        idx = idx.append(dti[-1:])
-        expect = Series(arr, index=idx)
+def test_resample_frame_basic():
+    df = tm.makeTimeDataFrame()
 
-        # GH2763 - return in put dtype if we can
-        result = g.agg(np.sum)
-        assert_series_equal(result, expect)
+    b = TimeGrouper('M')
+    g = df.groupby(b)
 
-        df = DataFrame(np.random.rand(len(dti), 10),
-                       index=dti, dtype='float64')
-        r = df.groupby(b).agg(np.sum)
+    # check all cython functions work
+    funcs = ['add', 'mean', 'prod', 'min', 'max', 'var']
```

Comment:
can you add to the list to parameterize (future PR is ok)

Reply:
I suspect this can be a fixture, as I see that list (or something similar to it) floating around in several places.